relay-server/src/managed/managed.rs

```diff
@@ -560,6 +582,14 @@ impl<T: Counted + fmt::Debug> fmt::Debug for Managed<T> {
     }
 }
 
+impl<T: Counted> Managed<Option<T>> {
+    /// Turns a managed option into an optional [`Managed`].
+    pub fn transpose(self) -> Option<Managed<T>> {
```

Comment:
Cool! Stuff like this is why I love Rust.

I need to think of something here eventually, I think ideally these combinators should be possible to implement without having to reach for `destructure`, just not sure how that'd look.